$NetBSD: patch-af,v 1.3 2005/09/27 11:25:37 dsainty Exp $

Since the programmer knows which algorithm should be used to program the
device, make use of this and select the new style programming process if
required.  This appears to be essential on the 16F628A.

--- lowlvl.c.orig	2005-09-18 19:19:52.000000000 +1200
+++ lowlvl.c	2005-09-18 19:21:42.000000000 +1200
@@ -674,7 +674,7 @@
 
   waitus( 5000 );
 
-  if (pconfig.alt) {      // modified 4/4/00
+  if (pconfig.alt || pic_device->algorithm) {
     send_cmnd( 0x02 );
     waitus( 1 );
     send_data ( 0x3FFF );
